src/java/neroxis/map/SCMap.java

```diff
@@ -376,6 +382,266 @@ public void addCubeMap(CubeMap cubeMap) {
         cubeMaps.add(cubeMap);
     }
 
+    public void resize(int resizeCurrentMapContentTo, int newMapBoundsSize, Vector2f locToPutCenterOfCurrentMapContent, float heightMultiplier) {
```

Comment:
this function is doing too much now resizing has nothing to do with height multiplication

Reply:
Resizing has a lot to do with height multiplication.
If you resize a map and you don't multiply the height accordingly, then the slopes will be wrong.  For example, if you take a regular 20 x 20 map and resize it to 5 x 5, you will get many impassable areas that should be passable, unless you scale/multiply the height accordingly.  Conversely, if you take a small map and resize it to be bigger, you will get gentler slopes (which can make previously impassable terrain instead be passable), unless you scale/multiply the height accordingly.